src/core/xconnection.rs

```diff
@@ -314,7 +337,7 @@ pub trait XConn {
     fn unmap_window(&self, id: WinId);
 
     /// Send an X event to the target window
-    fn send_client_event(&self, id: WinId, atom_name: &str) -> Result<()>;
+    fn send_client_event(&self, id: WinId, atom_name: &Atom) -> Result<()>;
```

Comment:
The public API of `XConn` should not be changing as part of this PR: usage of hard coded strings internally can (and should) be updated, but these methods need to be able to accept arbitrary strings in order to support end users requesting atoms not currently in the Enum defined above. Penrose is not fully supporting all EWMH atoms directly yet (and may never do so) but if users wish to work with atoms directly we need to be able to support that.

Reply:
Okay, that makes sense.